spyder/widgets/ipythonconsole/shell.py

```diff
@@ -220,27 +223,49 @@ def clear_console(self):
         else:
             self.execute("%clear")
 
-    def reset_namespace(self, force=False):
+    def reset_namespace(self, warning=None, silent=True):
```

Comment:
I think this is easier to understand if we make `warning=False` by default.

Reply:
I agree it makes the code clearer. However, `warning` will then always be `False` when `reset_namespace` is called from the console context menu. Therefore, I created a new method called `_reset_namespace` that is now used as a slot for the "reset namespace" action from the context menu instead of connecting it directly to the `reset_namespace` method.